### ISSUE TITLE
Enforce semantics role checks on required parameters

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#161346]
### Use case

After this pr https://github.com/flutter/flutter/pull/161260
we will be backfilling all the missing role in http://flutter.dev/go/semantics-roles, this will add a bunch of new ways people can get it wrong when assigning the new role to their widget. to help reduce the chaos we should also implement a way to enforce semantics role requirements. See [this doc](https://docs.google.com/document/d/1WOEraK9VIh9dzpJ3mrXgenzfbpRxeISNAdgH8Sh80a4/edit?tab=t.0#heading=h.j9ifh1mhb7b0)

For example, if user assigns a tab role, they needs to have a ontap if it is enabled etc.

### Proposal

implement the checks

**Comment by chunhtai** [comment]
done https://github.com/flutter/flutter/pull/162290#event-16115534354
